src/core/net/tcp6.cpp

```diff
@@ -163,25 +232,48 @@ Error Tcp::Endpoint::Deinitialize(void)
     SuccessOrExit(error = tcp.mEndpoints.Remove(*this));
     SetNext(nullptr);
 
+    SuccessOrExit(error = Abort());
+
 exit:
     return error;
 }
 
 uint8_t Tcp::Endpoint::TimerFlagToIndex(uint8_t aTimerFlag)
 {
-    OT_UNUSED_VARIABLE(aTimerFlag);
-    /*
-     * TODO: Convert from the timer flag provided by TCPlp to the index in
-     * our timers array.
-     */
-    return 0;
+    switch (aTimerFlag)
+    {
+    case TT_DELACK:
+        return kTimerDelack;
+    case TT_REXMT:
+        /* fallthrough intentional */
+    case TT_PERSIST:
+        return kTimerRexmtPersist;
+    case TT_KEEP:
+        return kTimerKeep;
+    case TT_2MSL:
+        return kTimer2Msl;
+    default:
+        OT_ASSERT(false);
+        return 0; /* prevents a compiler warning */
+    }
 }
 
 bool Tcp::Endpoint::IsTimerActive(uint8_t aTimerIndex)
 {
-    OT_UNUSED_VARIABLE(aTimerIndex);
-    /* TODO: Check whether TCPlp has marked this timer as active. */
-    return false;
+    OT_ASSERT(aTimerIndex < kNumTimers);
+    switch (aTimerIndex)
+    {
+    case kTimerDelack:
+        return tcp_timer_active(&mTcb, TT_DELACK);
+    case kTimerRexmtPersist:
+        return tcp_timer_active(&mTcb, TT_REXMT) || tcp_timer_active(&mTcb, TT_PERSIST);
+    case kTimerKeep:
+        return tcp_timer_active(&mTcb, TT_KEEP);
+    case kTimer2Msl:
+        return tcp_timer_active(&mTcb, TT_2MSL);
+    default:
+        return false;
```

Comment:
Avoid multiple `return` statements? (single `return` at the end)

Reply:
OK. Fixed in the latest update.